taxi/tests/test_manufacturer_views.py

```diff
@@ -0,0 +1,104 @@
+# flake8: noqa: E731
+
+from django.contrib.auth import get_user_model
+from django.test import TestCase
+from django.urls import reverse
+
+from taxi.models import Manufacturer
+
+MANUFACTURER_URL = reverse("taxi:manufacturer-list")
+MANUFACTURER_CREATE_URL = reverse("taxi:manufacturer-create")
+MANUFACTURER_UPDATE_URL = lambda pk: reverse(
+    "taxi:manufacturer-update", args=[pk]
+)
+MANUFACTURER_DELETE_URL = lambda pk: reverse(
+    "taxi:manufacturer-delete", args=[pk]
+)
+
+
+class PublicManufacturerTests(TestCase):
+    def test_login_required(self):
+        response = self.client.get(MANUFACTURER_URL)
+        self.assertNotEqual(response.status_code, 200)
+
+        response = self.client.get(MANUFACTURER_CREATE_URL)
+        self.assertNotEqual(response.status_code, 200)
+
+
+class PrivateManufacturerTests(TestCase):
+    def setUp(self) -> None:
+        self.user = get_user_model().objects.create_user(
+            username="test",
+            password="test1223",
+        )
+        self.client.force_login(self.user)
+        self.manufacturer = Manufacturer.objects.create(
+            name="Test Manufacturer",
+            country="US",
+        )
+
+    def test_search_manufacturer_by_username(self):
+        """Test searching manufacturer by name"""
```

Comment:
The docstring for `test_search_manufacturer_by_username` is misleading. It should mention searching by 'name' instead of 'username'.